conventions/src/main/kotlin/otel.java-conventions.gradle.kts

```diff
@@ -80,7 +80,8 @@ tasks.withType<JavaCompile>().configureEach {
         )
       )
       val defaultJavaVersion = otelJava.maxJavaVersionSupported.getOrElse(DEFAULT_JAVA_VERSION).majorVersion.toInt()
-      if (Math.max(otelJava.minJavaVersionSupported.get().majorVersion.toInt(), defaultJavaVersion) >= 21) {
+      if (Math.max(otelJava.minJavaVersionSupported.get().majorVersion.toInt(), defaultJavaVersion) >= 21 &&
```

Comment:
I doubt that this is the best way to fix this. My guess would be that this can be fixed with changing graal to use java 21 in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/eb8748b65b17a5d8efde381bc5294247a6b2134d/.github/workflows/reusable-native-tests.yml#L19 If that does not work then I'd try limiting java version to 17 with https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/eb8748b65b17a5d8efde381bc5294247a6b2134d/conventions/src/main/kotlin/io/opentelemetry/instrumentation/gradle/OtelJavaExtension.kt#L13

Reply:
let me try :smile: